foca/foca.py

```diff
@@ -0,0 +1,57 @@
+"""FOCA class definition"""
+
+from foca.api.register_openapi import (register_openapi, OpenAPIConfig)
+from foca.config.config_parser import (get_conf, get_conf_type)
+from foca.config.log_config import configure_logging
+from foca.database.register_mongodb import register_mongodb
+from foca.errors.errors import register_error_handlers
+from foca.factories.connexion_app import create_connexion_app
+from foca.security.cors import enable_cors
+from connexion import App
+from foca.config.config_handler import Config
+from typing import Iterable
+
+
+class FOCA():

```

Comment:
I know I said class, but thinking about it more on the weekend, I think this should really be just a function because we have too much code for a snappy class, so here I think instantiating a class both violates OOP philosophy (such that everything relevant to the class should be in the class) and Python conventions (where class code shouldn't really be split up over multiple modules). And while there are ways to achieve a more Pythonic way of dealing with this (in particular a sub-classing approach; see for example here: https://stackoverflow.com/questions/47561840/python-how-can-i-separate-functions-of-class-into-multiple-files), at least at the moment I don't really see too much benefit of that, given that the only data binding we have is in the config, which is also attached to the app, meaning that the only thing that we really _need_ to pass between function calls is the app object (which we can easily return rather then update the corresponding instance variable), even if we may choose to pass config params separately (e.g., in the new `foca.api.register_openapi` module) so as to make the functions more understandable and general.

So I would propose that instead of defining a class hear, just define a function `foca()` (not capitalized, according to Python naming conventions). It shouldn't affect the solution too much, in fact it should make it easier (no need for `__iinit__()` and `__new__()`).